### ISSUE TITLE
Feature modification: alter input checks and errors

### ISSUE THREAD

**Issue by lizlaw** [prioritizr/prioritizr#95]
Hi, 
I'm not sure how much of a hassle it might be, and it is a low priority suggestion... but from my end it would be easier if it wouldn't throw errors if, for example, there is a lockin specified, but there is no cell selected as a lockin. A warning would be fine. So long as it still computes ok! This is because sometimes in some of the scenario loops I am creating I sometimes end up with no lockin (e.g. if looping over sub-regions to do a scenario where planning is by sub-region, rather the region as a whole - I'm locking in existing protected areas, but these are not extant in all provinces).  I can easily get around it by specifying an 'if' condition, but might be easier and cleaner if i didn't have to do this. 
As I said, not a high priority! 
Cheers muchly,
Liz

**Comment by jeffreyhanson** [comment]
Yeah, I can see why it would be useful to have this functionality when looping over different inputs. I worry that implementing this functionality might confuse users and it might conflict with other functions in the sense that most of the functions in _prioritizr_ are designed to fail hard when invalid inputs are supplied. I think one of the strengths of _prioritizr_ is that it provides a consistent interface, so if we were to implement this functionality for the locked in constraints then we would have to make each other function (e.g. other constraints, targets, penalty functions) accept "empty" inputs and throw a warning (and perhaps generalize to empty list, matrices, spatial data types). I think this would be a lot of work, and personally I really like the fact that _prioritizr_ tells me when I have supplied invalid and empty input because most of the time this is a mistake on my part when data wrangling. What do you think?
